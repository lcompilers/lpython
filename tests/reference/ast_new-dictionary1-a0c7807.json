{
    "basename": "ast_new-dictionary1-a0c7807",
    "cmd": "lpython --show-ast --indent --new-parser --no-color {infile} -o {outfile}",
    "infile": "tests/parser/dictionary1.py",
    "infile_hash": "68922e77dbabe06c271463150653684880d3c28d88caa44109b2e202",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_new-dictionary1-a0c7807.stdout",
    "stdout_hash": "dfae0ace6e4267478ecfd053f2be4a56fddf1f635099b4f6695d250b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}