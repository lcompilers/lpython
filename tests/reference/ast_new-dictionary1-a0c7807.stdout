(Module
    [(FunctionDef
        test
        ([]
        []
        []
        []
        []
        []
        [])
        [(Assign
            [(Name
                x
                Store
            )]
            (Dict
                []
                []
            )
            ()
        )
        (Assign
            [(Name
                dict
                Store
            )]
            (Dict
                [(ConstantStr
                    "brand"
                    ()
                )
                (ConstantStr
                    "electric"
                    ()
                )
                (ConstantStr
                    "year"
                    ()
                )
                (ConstantStr
                    "colors"
                    ()
                )]
                [(ConstantStr
                    "Ford"
                    ()
                )
                (ConstantBool
                    .false.
                    ()
                )
                (ConstantInt
                    1964
                    ()
                )
                (List
                    [(ConstantStr
                        "red"
                        ()
                    )
                    (ConstantStr
                        "white"
                        ()
                    )]
                    Load
                )]
            )
            ()
        )
        (Assign
            [(Name
                num
                Store
            )]
            (Dict
                [(ConstantInt
                    1
                    ()
                )
                (ConstantInt
                    2
                    ()
                )]
                [(ConstantInt
                    2
                    ()
                )
                (ConstantInt
                    3
                    ()
                )]
            )
            ()
        )
        (AnnAssign
            (Name
                y
                Store
            )
            (Subscript
                (Name
                    dict
                    Load
                )
                (Tuple
                    [(Name
                        str
                        Load
                    )
                    (Name
                        i32
                        Load
                    )]
                    Load
                )
                Load
            )
            ()
            1
        )
        (Assign
            [(Name
                y
                Store
            )]
            (Dict
                [(ConstantStr
                    "a"
                    ()
                )
                (ConstantStr
                    "b"
                    ()
                )]
                [(UnaryOp
                    USub
                    (ConstantInt
                        1
                        ()
                    )
                )
                (UnaryOp
                    USub
                    (ConstantInt
                        2
                        ()
                    )
                )]
            )
            ()
        )
        (Assign
            [(Name
                y
                Store
            )]
            (Dict
                [(ConstantStr
                    "a"
                    ()
                )
                (ConstantStr
                    "b"
                    ()
                )]
                [(UnaryOp
                    USub
                    (ConstantInt
                        1
                        ()
                    )
                )
                (UnaryOp
                    USub
                    (ConstantInt
                        2
                        ()
                    )
                )]
            )
            ()
        )
        (Assign
            [(Subscript
                (Name
                    y
                    Load
                )
                (ConstantStr
                    "a"
                    ()
                )
                Store
            )]
            (ConstantInt
                123
                ()
            )
            ()
        )
        (Assign
            [(Name
                a
                Store
            )]
            (Dict
                [(ConstantStr
                    "a"
                    ()
                )
                (ConstantStr
                    "b"
                    ()
                )]
                [(ConstantInt
                    1
                    ()
                )
                (ConstantInt
                    2
                    ()
                )]
            )
            ()
        )
        (Assign
            [(Name
                x
                Store
            )]
            (Dict
                [(ConstantStr
                    "markdown2.tpl"
                    ()
                )]
                [(ConstantStr
                    "          Text\n          "
                    ()
                )]
            )
            ()
        )
        (Assign
            [(Name
                x
                Store
            )]
            (Dict
                [(ConstantNone
                    ()
                )
                (Name
                    y
                    Load
                )
                (ConstantNone
                    ()
                )
                (ConstantNone
                    ()
                )]
                [(Name
                    x
                    Load
                )
                (ConstantStr
                    "Text"
                    ()
                )
                (Name
                    z
                    Load
                )
                (ConstantStr
                    "None"
                    ()
                )]
            )
            ()
        )]
        []
        ()
        ()
    )
    (Expr
        (Call
            (Name
                test
                Load
            )
            []
            []
        )
    )]
    [(TypeIgnore
        0
        ""
    )]
)
